plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/DecoderType.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch.decoders;
+
+import io.trino.spi.type.StandardTypes;
+import io.trino.spi.type.Type;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Arrays.stream;
+import static java.util.function.Function.identity;
+
+public enum DecoderType
+{
+    ARRAY(StandardTypes.ARRAY, true),
+    BIGINT(StandardTypes.BIGINT, true),
+    BOOLEAN(StandardTypes.BOOLEAN, true),
+    DOUBLE(StandardTypes.DOUBLE, true),
+    INTEGER(StandardTypes.INTEGER, true),
+    IP_ADDRESS(StandardTypes.IPADDRESS, true),
+    RAW_JSON(StandardTypes.VARCHAR, false),
+    REAL(StandardTypes.REAL, true),
+    ROW(StandardTypes.ROW, true),
+    SMALLINT(StandardTypes.SMALLINT, true),
+    TIMESTAMP(StandardTypes.TIMESTAMP, true),
+    TINYINT(StandardTypes.TINYINT, true),
+    VARBINARY(StandardTypes.VARBINARY, true),
+    VARCHAR(StandardTypes.VARCHAR, true);
+
+    private static final Map<String, DecoderType> DECODER_TYPE_BY_CLASS_TYPE = stream(values())
+            .filter(DecoderType::isDefaultType)
+            .collect(toImmutableMap(DecoderType::getTypeBaseName, identity()));
+
+    private final String typeBaseName;
+    private final boolean defaultType;
+
+    DecoderType(String typeBaseName, boolean defaultType)
+    {
+        this.typeBaseName = typeBaseName;
+        this.defaultType = defaultType;
+    }
+
+    public static DecoderType of(Type type)
+    {
+        return of(type.getBaseName());
+    }
+
+    public static DecoderType of(String typeBaseName)
+    {
+        if (typeBaseName.startsWith(StandardTypes.TIMESTAMP)) {
+            return TIMESTAMP;
+        }
+
+        checkArgument(
```

Comment:
I don't think this provides much value and just complicates the structure of this class. E.g., it requires the "default type" flag and other machinery. I would just do the check in `getDecoderContext`. If there's no decoder, throw an error there.

Reply:
@martint As part of #8461, it should be taken care of.